src/options.c

```diff
@@ -585,7 +585,9 @@ void opt_init(char *name, int argc, char **argv, int show_usage)
 #if HAVE_OPENCL
 	if (options.format && strcasestr(options.format, "opencl") &&
 	    (options.flags & FLG_FORK) && options.acc_devices->count == 0) {
-		list_add(options.acc_devices, "all");
+		list_add(options.acc_devices, "gpu");
+		/* Set a flag that JtR has changed the value of --devices. */
+		default_device_selected = 1;
 	}
```

Comment:
What if we only have accelerators, or only CPUs? Perhaps after adding gpu's you could check count again, if still zero try next preferable (accel), and then again (CPU).

Reply:
OpenCL CPUs are in the past (the new AMD driver do not support CPUs). So, only Intel does(?)

> Perhaps after adding gpu's you could check count again, if still zero try next preferable (accel), and then again (CPU).

This is feasible. And it will happens only in the **default** situation.

Reply:
The fact AMD stopped supporting them in their GPU drivers doesn't mean it's the past: Intel has good drivers now, pocl has it, beignet, macOS... Perhaps AMD plans to release a separate CPU driver/runtime.

Reply:
Every time I tried to find any CPU implementation (to do CI), I was disappointed. Anyway, it is not important.